linked_list/linked_list.py

```diff
@@ -86,11 +175,41 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)

```

Comment:
your time/space complexity is correct! But how could we solve this so that we don't need to make new data structures? How could we use a few pointers that keep track of the previous node, the current node, and the next node?